File: modules/go_workloads/src/string_churn.go
// O(n²) string concatenation: strings are immutable, so every round
// allocates a fresh string and copies the whole accumulator — high
// transient allocation rate, tiny live heap at any instant.

package go_workloads

func Concat(pieces int, piece string) string {
	acc := ""
	for i := 0; i < pieces; i++ {
		acc += piece
	}
	return acc
}

File: modules/java_workloads/src/StringChurn.java
package monorepo.workloads;

/**
 * O(n²) string concatenation: strings are immutable, so every round
 * allocates a fresh string and copies the whole accumulator — high
 * transient allocation rate, tiny live heap at any instant.
 */
public final class StringChurn {
    private StringChurn() {
    }

    public static String concat(int pieces, String piece) {
        String acc = "";
        for (int i = 0; i < pieces; i++) {
            // String.concat, not +: same fresh-allocation-and-copy churn,
            // without javac's invokedynamic concat rewriting it.
            acc = acc.concat(piece);
        }
        return acc;
    }
}
